From 8831ded554a1ff3b1a84c02113dad55241642f16 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:51:17 -0400
Subject: [tools] lib/traceevent: Add global QUIET_CC_FPIC build output

Message-id: <1408953210-26343-190-git-send-email-jolsa@redhat.com>
Patchwork-id: 88960
O-Subject: [PATCH RHEL7.1 BZ1133083 189/322] tools lib traceevent: Add global QUIET_CC_FPIC build output
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit bdebbacd421184234b5551c468e53956a9b6edf5
Author: Jiri Olsa <jolsa@redhat.com>
Date: Thu Dec 19 14:42:03 2013 +0100

description
===========
Adding global QUIET_CC_FPIC build output variable and getting rid of
local print_fpic_compile and print_plugin_obj_compile.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/Makefile b/tools/lib/traceevent/Makefile
index e852a8d..24203cc 100644
--- a/tools/lib/traceevent/Makefile
+++ b/tools/lib/traceevent/Makefile
@@ -152,33 +152,21 @@ override CFLAGS += $(udis86-flags) -D_GNU_SOURCE
 
 ifeq ($(VERBOSE),1)
   Q =
-  print_fpic_compile =
   print_shared_lib_compile =
-  print_plugin_obj_compile =
   print_plugin_build =
   print_install =
 else
   Q = @
-  print_fpic_compile =  echo '  CC FPIC  '$(OBJ);
   print_shared_lib_compile = echo '  BUILD    SHARED LIB '$(OBJ);
-  print_plugin_obj_compile = echo '  CC FPIC  '$(OBJ);
   print_plugin_build =  echo '  BUILD    PLUGIN '$(OBJ);
   print_static_lib_build = echo '  BUILD    STATIC LIB '$(OBJ);
   print_install =  echo '  INSTALL  '$1;
 endif
 
-do_fpic_compile =     \
- ($(print_fpic_compile)    \
- $(CC) -c $(CFLAGS) $(EXT) -fPIC $< -o $@)
-
 do_compile_shared_library =   \
  ($(print_shared_lib_compile)  \
  $(CC) --shared $^ -o $@)
 
-do_compile_plugin_obj =    \
- ($(print_plugin_obj_compile)  \
- $(CC) -c $(CFLAGS) -fPIC -o $@ $<)
-
 do_plugin_build =    \
  ($(print_plugin_build)   \
  $(CC) $(CFLAGS) -shared -nostartfiles -o $@ $<)
@@ -236,10 +224,10 @@ libtraceevent.a: $(PEVENT_LIB_OBJS)
 plugins: $(PLUGINS)
 
 $(PEVENT_LIB_OBJS): %.o: $(src)/%.c TRACEEVENT-CFLAGS
- $(Q)$(do_fpic_compile)
+ $(QUIET_CC_FPIC)$(CC) -c $(CFLAGS) $(EXT) -fPIC $< -o $@
 
 $(PLUGIN_OBJS): %.o : $(src)/%.c
- $(Q)$(do_compile_plugin_obj)
+ $(QUIET_CC_FPIC)$(CC) -c $(CFLAGS) -fPIC -o $@ $<
 
 $(PLUGINS): %.so: %.o
  $(Q)$(do_plugin_build)
diff --git a/tools/scripts/Makefile.include b/tools/scripts/Makefile.include
index e4cfbed..8abbef1 100644
--- a/tools/scripts/Makefile.include
+++ b/tools/scripts/Makefile.include
@@ -61,6 +61,7 @@ QUIET_SUBDIR1  =
 ifneq ($(findstring $(MAKEFLAGS),s),s)
   ifneq ($(V),1)
  QUIET_CC       = @echo '  CC       '$@;
+ QUIET_CC_FPIC  = @echo '  CC FPIC  '$@;
  QUIET_AR       = @echo '  AR       '$@;
  QUIET_LINK     = @echo '  LINK     '$@;
  QUIET_MKDIR    = @echo '  MKDIR    '$@;
-- 
1.7.1